src/NationalInstruments.Analyzers/Style/LongChainDottedInvocationAnalyzer.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NationalInstruments.Analyzers.Properties;
+using NationalInstruments.Analyzers.Utilities;
+using NationalInstruments.Analyzers.Utilities.Extensions;
+
+namespace NationalInstruments.Analyzers.Style
+{
+    /// <summary>
+    /// Enforces the convention -
+    /// https://github.com/ni/csharp/blob/main/docs/Coding-Conventions.md#f34-%EF%B8%8F-consider-splitting-long-chains-of-dotted-methodproperty-invocations
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class LongChainDottedInvocationAnalyzer : NIDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "NI1017";
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            new LocalizableResourceString(nameof(Resources.NI1017_Title), Resources.ResourceManager, typeof(Resources)),
+            new LocalizableResourceString(nameof(Resources.NI1017_Message), Resources.ResourceManager, typeof(Resources)),
+            Resources.CategoryNI,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecutionIf(IsRunningInProduction);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(AnalyzeExpressionStatementSyntax, SyntaxKind.ExpressionStatement);
+            context.RegisterSyntaxNodeAction(AnalyzeEqualsValueClauseSyntax, SyntaxKind.EqualsValueClause);
+            context.RegisterSyntaxNodeAction(AnalyzeArgumentSyntax, SyntaxKind.Argument);
+            context.RegisterSyntaxNodeAction(AnalyzeArrowExpressionClauseSyntax, SyntaxKind.ArrowExpressionClause);
```

Comment:
Since these individual syntax handlers don't actually do different things, you can just pass all the `SyntaxKind`s in the second parameter (it's `params SyntaxKind[]` I think)

Reply:
Thanks, I missed it. I will update.